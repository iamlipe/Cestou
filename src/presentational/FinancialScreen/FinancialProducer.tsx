import React from 'react';
import styled from 'styled-components/native';

export const FinancialProducer = () => {
  return (
    <StyledContainer>
      <StyledText>FinancialProducer</StyledText>
    </StyledContainer>
  );
};

const StyledContainer = styled.View``;
const StyledText = styled.Text``;
