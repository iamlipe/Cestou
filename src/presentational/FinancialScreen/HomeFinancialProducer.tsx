import React, {useEffect} from 'react';
import styled, {useTheme} from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {useReduxSelector} from '@/hooks/useReduxSelector';
import {useReduxDispatch} from '@/hooks/useReduxDispatch';
import {GET_PRODUCER} from '@/store/slices/producerSlice';

import IconPiggyBank from '@/assets/svgs/piggy-bank.svg';

import Header from '@/components/Header';
import ButtonRedirect from '@/components/ButtonRedirect';

export const HomeFinancialProducer = () => {
  const {auth} = useReduxSelector(state => state.user);
  const {producer} = useReduxSelector(state => state.producer);
  const theme = useTheme();
  const dispatch = useReduxDispatch();

  useEffect(() => {
    if (auth?.id) {
      dispatch(GET_PRODUCER({id: auth?.id}));
    }
  }, [auth?.id, dispatch]);

  const renderPix = (key: string, iconName: string) => {
    <>
      <Icon name={iconName} size={20} color={theme.colors.PRIMARY_800} />
      <StyledText>{key}</StyledText>
    </>;
  };

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header title="Financeiro" welcome={false} />
      <StyledContent>
        <StyledPiggyBank />
        <StyledTitleBalance>Saldo dispon√≠vel</StyledTitleBalance>
        {producer?.balance ? (
          <StyledTextBalance>
            {`R$ ${producer?.balance.toLocaleString('pt-br', {
              style: 'currency',
              currency: 'BRL',
            })}`}
          </StyledTextBalance>
        ) : (
          <StyledTextLoadingBalance>Carregando...</StyledTextLoadingBalance>
        )}
        <ButtonRedirect title="Configurar chave pix" onPress={() => null} />
        <ButtonRedirect title="Ver extrato" onPress={() => null} />

        {(producer?.cpfPix ||
          producer?.emailPix ||
          producer?.phonePix ||
          producer?.randomPix) && (
          <>
            <StyledLine />
            <StyledTitlePix>Chaves cadastradas</StyledTitlePix>
            {producer?.cpfPix && renderPix(producer?.cpfPix, 'person')}
            {producer?.emailPix &&
              renderPix(producer?.emailPix, 'mail-outline')}
            {producer?.phonePix &&
              renderPix(producer?.phonePix, 'mobile-friendly')}
            {producer?.randomPix && renderPix(producer?.randomPix, 'vpn-key')}
          </>
        )}
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledPiggyBank = styled(IconPiggyBank)`
  align-self: center;
`;

const StyledTitleBalance = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGER};
  color: ${({theme}) => theme.colors.GRAY_900};

  text-align: center;
  margin-top: 16px;
`;

const StyledTextBalance = styled(StyledTitleBalance)`
  font-size: ${({theme}) => theme.sizing.LARGEST};
  margin-bottom: 16px;
`;

const StyledTextLoadingBalance = styled.Text``;

const StyledLine = styled.View`
  width: 100%;
  height: 1px;
  align-self: center;
  background-color: ${({theme}) => theme.colors.GRAY_300};
  margin-bottom: 16px;
`;

const StyledTitlePix = styled.Text``;

const StyledText = styled.Text``;
