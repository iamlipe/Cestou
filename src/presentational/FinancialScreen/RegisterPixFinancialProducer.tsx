import React, {useState} from 'react';
import styled from 'styled-components/native';
import * as Yup from 'yup';
import {useForm} from 'react-hook-form';
import {
  GET_PRODUCER,
  RegisterPixRequest,
  REGISTER_PIX,
} from '@/store/slices/producerSlice';
import {yupResolver} from '@hookform/resolvers/yup';
import {useReduxDispatch} from '@/hooks/useReduxDispatch';
import {useReduxSelector} from '@/hooks/useReduxSelector';
import {useTranslation} from 'react-i18next';
import {translatePixType} from '@/helpers/translate';
import {SvgProps} from 'react-native-svg';

import IconPiggy from '@/assets/svgs/piggy.svg';

import Header from '@/components/Header';
import RadioForm from '@/components/RadioForm';
import InputForm from '@/components/InputForm';
import Button from '@/components/Button';
import Modal from '@/components/Modal';

export const RegisterPixFinancialProducer = () => {
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  const {auth} = useReduxSelector(state => state.user);
  const {isLoading} = useReduxSelector(state => state.producer);
  const {t} = useTranslation();
  const dispatch = useReduxDispatch();

  const schema = Yup.object().shape({
    pixType: Yup.string().required(t('error.required')),
    pixValue: Yup.string().required(t('error.required')),
  });

  const {
    control,
    handleSubmit,
    formState: {errors, isSubmitted},
  } = useForm<RegisterPixRequest>({
    resolver: yupResolver(schema),
  });

  function registerPix(data: RegisterPixRequest) {
    const pixType = translatePixType(data.pixType);

    if (pixType) dispatch(REGISTER_PIX({pixType, pixValue: data.pixValue}));
    if (auth?.id) dispatch(GET_PRODUCER({id: auth?.id}));
  }

  function handleModal(event: boolean) {
    setIsVisibleModal(event);
  }

  async function onSubmit(data: RegisterPixRequest) {
    registerPix(data);
    handleModal(true);
  }

  return (
    <StyledContainerScroll
      showsVerticalScrollIndicator={false}
      testID="register-pix-producer-screen">
      <Header
        title={t('text.screenRegisterPixFinancialProducer.headerTitle')}
        welcome={false}
      />
      <StyledContent>
        <StyledText>
          {t('text.screenRegisterPixFinancialProducer.instructionOne')}
        </StyledText>
        <RadioForm
          name="pixType"
          control={control}
          options={[
            t('option.typePixCPF'),
            t('option.typePixPhone'),
            t('option.typePixEmail'),
            t('option.typePixRandom'),
          ]}
          error={isSubmitted ? errors.pixType?.message : ''}
          type="withLine"
        />
        <StyledText>
          {t('text.screenRegisterPixFinancialProducer.instructionTwo')}
        </StyledText>
        <InputForm
          name="pixValue"
          control={control}
          error={isSubmitted ? errors.pixValue?.message : ''}
          label={t('label.pixValue')}
        />
        <Button
          testID="submit-button"
          title={t('button.registerPix')}
          loading={isLoading}
          onPress={handleSubmit(onSubmit)}
        />

        {!isLoading && isVisibleModal && (
          <Modal
            title={t('modal.titleRegisterPix')}
            icon={IconPiggy as React.FC<SvgProps>}
            onClose={() => handleModal(false)}
            justMessage
          />
        )}
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLEST};
  color: ${({theme}) => theme.colors.GRAY_900};

  margin-bottom: 16px;
`;
