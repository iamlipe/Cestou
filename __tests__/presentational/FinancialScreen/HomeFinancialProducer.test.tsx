import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import {HomeFinancialProducer} from '@/presentational/FinancialScreen/HomeFinancialProducer';

describe('HomeFinancialProducer', () => {
  test('should render screen correctly', () => {
    renderWithThemeProvider(<HomeFinancialProducer />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<HomeFinancialProducer />);
  });

  test('should render piggy bank', () => {
    const homeFinancialProducer = renderWithThemeProvider(
      <HomeFinancialProducer />,
    );

    const piggyBank = homeFinancialProducer.getByTestId('icon-piggy-bank');

    expect(piggyBank).toBeTruthy();
  });
});
