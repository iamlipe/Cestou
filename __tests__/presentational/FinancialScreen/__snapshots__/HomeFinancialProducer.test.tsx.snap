// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeFinancialProducer should matches snapshot 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "minHeight": "100%",
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6CCD91",
              "flexDirection": "row",
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": "5%",
              "paddingRight": "5%",
              "paddingTop": 0,
              "width": "100%",
            },
          ]
        }
        type="normal"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "space-between",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#1A1A1A",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 47,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Financeiro
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": 24,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 24,
            "width": "90%",
          },
        ]
      }
    >
      <SvgMock
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
          ]
        }
        testID="icon-piggy-bank"
      />
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 43,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        Saldo disponível
      </Text>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Carregando...
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C1E9CF",
            "borderColor": "#6CCD91",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 49,
            "justifyContent": "space-between",
            "marginBottom": 16,
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontFamily": "SourceSansPro-Regular",
                "fontSize": 27,
              },
            ]
          }
        >
          Configurar chave pix
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#C1E9CF",
            "borderColor": "#6CCD91",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 49,
            "justifyContent": "space-between",
            "marginBottom": 16,
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontFamily": "SourceSansPro-Regular",
                "fontSize": 27,
              },
            ]
          }
        >
          Ver extrato
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
